libs/scripts/taxonomy.js

```diff
@@ -0,0 +1,249 @@
+    return taxonomy.products[t];
+  }
+  if (!category) {
+    return taxonomy.topics[t];
+  }
+  if (isProduct(category)) {
+    return taxonomy.products[t];
+  }
+  return taxonomy.topics[t];
+})[0];
+
+async function fetchTaxonomy(target) {
+  return fetch(target).then((response) => response.json());
+}
+
+function parseTaxonomyJson(data, root) {
+  return data?.reduce((taxonomy, row) => {
+    const level3 = removeLineBreaks(row[TAXONOMY_FIELDS.level3]);
+    const level2 = removeLineBreaks(row[TAXONOMY_FIELDS.level2]);
+    const level1 = removeLineBreaks(row[TAXONOMY_FIELDS.level1]);
+
+    // eslint-disable-next-line no-nested-ternary
+    const level = level3 ? LEVEL_INDEX.level3
+      : (level2 ? LEVEL_INDEX.level2
+        : LEVEL_INDEX.level1);
+
+    const name = level3 || level2 || level1;
+
+    const category = row[TAXONOMY_FIELDS.type]?.trim().toLowerCase() || INTERNALS;
+
+    // skip duplicates
+    if (!isProduct(category) && taxonomy.topics[name]) return taxonomy;
+    if (isProduct(category) && taxonomy.products[name]) return taxonomy;
+
+    const link = [row[TAXONOMY_FIELDS.link]].reduce((_url) => {
+      if (_url) {
+        const u = new URL(_url);
+        const current = new URL(window.location.href);
+        return `${current.origin}${u.pathname}`;
+      }
+
+      return `${root}/${generateUri(name)}`;
+    }, null);
+
+    const hidden = !!row[TAXONOMY_FIELDS.hidden]?.trim();
+    const skipMeta = !!row[TAXONOMY_FIELDS.excludeFromMetadata]?.trim();
+
+    const item = {
+      name,
+      level,
+      link,
+      category,
+      hidden,
+      skipMeta,
+    };
+
+    if (isProduct(category)) {
+      taxonomy.products[name] = item;
+    } else {
+      taxonomy.topics[name] = item;
+    }
+
+    if (!taxonomy.categories[item.category]) {
+      taxonomy.categories[item.category] = [];
+    }
+
+    if (taxonomy.categories[item.category].indexOf(name) === -1) {
+      taxonomy.categories[item.category].push(item.name);
+    }
+
+    const children = isProduct(category) ? taxonomy.productChildren : taxonomy.topicChildren;
+
+    if (level3 && !children[level2]) {
+      children[level2] = [];
+    } else if (level3 && children[level2].indexOf(level3) === -1) {
+      children[level2].push(level3);
+    }
+
+    if (level2 && !children[level1]) {
+      children[level1] = [];
+    } else if (level2 && children[level1].indexOf(level2) === -1) {
+      children[level1].push(level2);
+    }
+
+    return taxonomy;
+  }, {
+    topics: {},
+    products: {},
+    categories: {},
+    topicChildren: {},
+    productChildren: {},
+  });
+}
+
+/**
+ * Returns the taxonomy object
+ * @param {string} lang Language of the taxonomy
+ * @param {*} url URL to use to load the taxonomy
+ * @returns {object} The taxonomy object
+ */
+export default async (lang, url) => {
+  const prefix = lang ? `/${lang}` : '';
+  const root = `${prefix}/topics`;
+  const target = url || `${root}/taxonomy.json`;
```

Comment:
You should use `config.locale.contentRoot` for this.

Reply:
Updated 👍🏽